src/server/api/controllers/port.controller.js

```diff
@@ -31,12 +30,36 @@ const createPort = async ({ body }) => {
       name
     });
 
+  const suggested_route = await knex
+    .from("suggested_routes")
+    .select("id")
+    .where({
+      id: suggested_route_id
+    });
+
   if (ports.length !== 0) {
     throw new HttpError("Bad request", "port already exists!", 409);
   } else {
-    return knex("ports").insert({
-      name: name
-    });
+    return knex("ports")
+      .insert({
+        name: name
+      })
+      .then(portID => {
+        //if (suggested_route.length > 0 && waypoints) {
+        if (waypoints) {
+          return knex("waypoints").insert({
+            port_id: portID[0],
+            suggested_route_id: suggested_route[0].id,
```

Comment:
We want to get rid of this one as soon as my waypoints PR is merged.

Reply:
Comment resolved.